app/uk/gov/hmrc/excisemovementcontrolsystemapi/filters/MovementFilter.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.excisemovementcontrolsystemapi.filters
+
+import uk.gov.hmrc.excisemovementcontrolsystemapi.repository.model.Movement
+
+sealed trait Filter {
+  def filter(movements: Seq[Movement]): Seq[Movement]
+}
+
+case class FilterErn(ern: Option[String] = None) extends Filter {
+  def filter(movements: Seq[Movement]): Seq[Movement] = {
+    ern.fold[Seq[Movement]](movements)(e =>
+      movements.filter(o => o.consignorId.equals(e))
+    )
+  }
+
+}
+case class FilterLrn(lrn: Option[String] = None) extends Filter {
+  def filter(movements: Seq[Movement]): Seq[Movement] = {
+    lrn.fold[Seq[Movement]](movements)(l =>
+      movements.filter(o => o.localReferenceNumber.equals(l))
+    )
+  }
+}
+
+case class FilterArc(arc: Option[String] = None) extends Filter {
+  def filter(movements: Seq[Movement]): Seq[Movement] = {
+    arc.fold[Seq[Movement]](movements)(a =>
+      movements.filter(o => o.administrativeReferenceCode.contains(a))
+    )
+  }
+}
+
+class FilterEmpty extends Filter {
```

Comment:
Could rename this to e.g. FilterNothing as filterEmpty suggests it's filtering out empty things or something perhaps

Reply:
what about FilterNone?